README.adoc

```diff
@@ -35,38 +35,178 @@ image:https://coveralls.io/repos/github/aerogear/unifiedpush-operator/badge.svg?
-unifiedpush-operator in your OpenShift cluster as follows:
+|===
+|https://golang.org/doc/install[Install Go]
+|https://github.com/golang/go/wiki/SettingGOPATH[Ensure the $GOPATH environment variable is set]
+|https://golang.github.io/dep/docs/installation.html[Install the dep package manager]
+|https://github.com/operator-framework/operator-sdk#quick-start[Install Operator-SDK]
+|https://kubernetes.io/docs/tasks/tools/install-kubectl/#install-kubectl[Install kubectl]
+|===
 
-....
-kubectl create namespace unifiedpush
-kubectl create -n unifiedpush -f deploy/service_account.yaml
-kubectl create -n unifiedpush -f deploy/role.yaml
-kubectl create -n unifiedpush -f deploy/role_binding.yaml
-kubectl create -n unifiedpush -f deploy/crds/push_v1alpha1_unifiedpushserver_crd.yaml
-kubectl create -n unifiedpush -f deploy/crds/push_v1alpha1_pushapplication_crd.yaml
-kubectl create -n unifiedpush -f deploy/crds/push_v1alpha1_androidvariant_crd.yaml
-kubectl create -n unifiedpush -f deploy/crds/push_v1alpha1_iosvariant_crd.yaml
-kubectl create -n unifiedpush -f deploy/operator.yaml
-....
+== Getting Started
+
+=== Cloning the repository
+
+By the following commands you will create a local directory and clone this project.
+
+[source,shell]
+----
+$ git clone git@github.com:aerogear/unifiedpush-operator.git $GOPATH/src/github.com/aerogear/unifiedpush-operator
+----
+
+=== Minishift installation and setup
+
+https://docs.okd.io/latest/minishift/getting-started/installing.html[Install Minishift] then install Operators on it by running the following commands.
+
+[source,shell]
+----
+# create a new profile to test the operator
+$ minishift profile set unifiedpush-operator
+
+# enable the admin-user add-on
+$ minishift addon enable admin-user
+
+# add insecure registry to download the images from docker
+$ minishift config set insecure-registry 172.30.0.0/16
+
+# start the instance
+$ minishift start
+----
+
+NOTE: The above steps are not required in OCP > 4 since the OLM and Operators came installed by default.
+
+=== Installing
+
+Use the following command to install the UnifiedPush Operator and Service in your OpenShift cluster as follows:
+
+[source,shell]
+----
+$ make install
+----
+
+NOTE: To install you need be logged in as a user with cluster privileges like the `system:admin` user. E.g. By using: `oc login -u system:admin`.
+
+=== Creating/Binding an app
+
+* Create a link:./deploy/crds/examples/push_v1alpha1_pushapplication_cr.yaml[PushApplication CR] as this example.
++
+The app name and description need to be specified into the link:./deploy/crds/examples/push_v1alpha1_pushapplication_cr.yaml[PushApplication CR] as follows.
++
+[source,yaml]
+----
+apiVersion: push.aerogear.org/v1alpha1
+kind: PushApplication
+metadata:
+  name: example-pushapplication
+spec:
+  description: 'An example push application to demonstrate the
+    unifiedpush-operator'
+----
++
+* Run the following command to bind the app to the Push Service
```

Comment:
Similar to above. I would change it to "... to create a PushApplication".

Reply:
done